src/main/java/emissary/util/shell/Executrix.java

```diff
@@ -173,12 +173,15 @@ public static byte[] readFile(final String theFileName, final int length) throws
 
         try (InputStream theStream = Files.newInputStream(Paths.get(theFileName))) {
             final int avail = theStream.available();
-            if ((length == -1) || (length >= avail)) {
-                theContent = new byte[avail];
-            } else {
-                theContent = new byte[length];
+            final int expected = length == -1 || length >= avail ? avail : length;
```

Comment:
The readFile(...) method depends on InputStream.available() to size the buffer that will contain the data from the file. The javadoc for InputStream.available() says, "It is never correct to use the return value of this method to allocate a buffer intended to hold all data in this stream." I would suggest either...
1. Don't use InputStream.available() and read either until the end of the stream (i.e. end of the file) or the passed in "length" has been reached.
2. Use RandomAccessFile. This allows one to get the actual length of the file in order to correctly size the buffer and then do a full read of the data into the buffer.